llpc/lower/llpcSpirvLowerGlobal.cpp

```diff
@@ -1861,6 +1870,30 @@ void SpirvLowerGlobal::storeOutputMember(Type *outputTy, Value *storeValue, Arra
   llvm_unreachable("Should never be called!");
 }
 
+// Find a new insertion point that dominates the current insertion point \p instA and a specified instruction use
+// \p useB.
+static Instruction *findNearestCommonDominator(Instruction *instA, Use &useB, DominatorTree &domTree) {
+  auto *instB = cast<Instruction>(useB.getUser());
+  BasicBlock *blockB = instB->getParent();
+  if (auto *phiB = dyn_cast<PHINode>(instB)) {
+    // Normal instructions cannot be inserted before a PHI node, so put them at the end of the corresponding predecessor
+    // block instead.
+    blockB = phiB->getIncomingBlock(useB.getOperandNo());
+    instB = blockB->getTerminator();
+  }
+
+  // Allow instA to be nullptr for ease of use.
+  if (!instA)
+    return instB;
+
+  BasicBlock *blockA = instA->getParent();
+  if (blockA == blockB)
+    return instA->comesBefore(instB) ? instA : instB;
+
+  BasicBlock *block = domTree.findNearestCommonDominator(blockA, blockB);
+  return block == blockA ? instA : block == blockB ? instB : block->getTerminator();
```

Comment:
Better to add a () for clearness
->
(block == blockB ? instB : block->getTerminator())